js/pokemon.js

```diff
@@ -357,17 +370,49 @@ function revealPokemon(correctlyGuessed, language) {
 
 function generateNewNumbers(force) {
 
-    if(force || (currentGen !== newGen)) {
-        var i=0, j;
-        
+    if(force || !sortCompareArrays(currentGen, newGen)) {
+
+        console.log("generating New Pokemon array")
+
         upcomingPokemon = new Array();
         upcomingPokemonArrayPos = 0;
-        
-        for(j=minPokemonNumber; j<=maxPokemonNumber; j++) {
-            upcomingPokemon[i] = j;
-            i++;
+        var i = 0;
```

Comment:
General note: I moved away from single character identifiers unless it is a one off loop and trivial to understand. The amount of times I get lost in nested loops because people use i, j and k variable names. Do the world a favour even if you call them apples and bananas it makes it easier to read.